client/guidestours/index.js

```diff
@@ -58,12 +58,18 @@ class GuidesTours extends Component {
 
 	next() {
 		const nextStepName = this.props.tourState.stepConfig.next;
-		this.props.nextGuidesTourStep( nextStepName );
+		this.props.nextGuidesTourStep( { stepName: nextStepName } );
 	}
 
-	quit() {
+	quit( options = {} ) {
 		this.currentTarget && this.currentTarget.classList.remove( 'guidestours__overlay' );
-		this.props.nextGuidesTourStep( null );
+		this.props.quitGuidesTour( Object.assign( {
+			stepName: this.props.tourState.stepName
+		}, options ) );
+	}
+
+	finish() {
+		this.quit( { finished: true } );
```

Comment:
Took the liberty of refactoring things here so that we don't repeat code in `finish`.